src/main/java/com/diffblue/deeptestutils/mock/MemberMatcher.java

```diff
@@ -0,0 +1,100 @@
+ */
+public final class MemberMatcher {
+
+    /**
+     * Private constructor to prevent instantiation of the utility class.
+     */
+    private MemberMatcher() {
+    }
+
+    /**
+     * Get a method matching the given name and parameter types.
+     * <p>
+     * We first call the MemberMatcher of PowerMock to try and find a
+     * matching method. We return it's result unless it finishes with the
+     * TooManyMethodsFoundException exception. In that case, we catch the
+     * exception and try to find the method ourselves as follows. First, we try
+     * to find a matching method in the provided class. If no matching method
+     * is found then we proceed to the super class continuing up the class
+     * hierarchy. A matching method is a method with the provided name and
+     * parameter types that exaclty match the provided ones or are less
+     * specific. If there is more than one matching method found in a class
+     * than the one with the most specific return type is chosen.
+     *
+     * @param declaringClass The class to search
+     * @param methodName The name of the desired method
+     * @param parameterTypes Types that can be fed as arguments to the method
+     *                      (can be more specific than the method's declared
+     *                       types)
+     * @return A matching method if such exists
+     * @throws MethodNotFoundException If a method cannot be found in the
+     *                      hierarchy.
+     */
+    public static Method method(final Class<?> declaringClass,
+                                final String methodName,
+                                final Class<?>... parameterTypes) {
+
+        try {
+            return org.powermock.api.support.membermodification.MemberMatcher
+                    .method(declaringClass, methodName, parameterTypes);
+        } catch (TooManyMethodsFoundException e) {
+
+            Class<?> currentClass = declaringClass;
+
+            while (currentClass != null) {
+                Method[] declaredMethods = currentClass.getDeclaredMethods();
+                LinkedList<Method> matchingMethods = new LinkedList<Method>();
+
+                for (Method method : declaredMethods) {
+                    boolean parametersMatch =
+                            WhiteboxImpl.checkIfParameterTypesAreSame(
+                                    method.isVarArgs(), parameterTypes,
+                                    method.getParameterTypes());
+                    if (methodName.equals(method.getName())
+                            && parametersMatch) {
+                        matchingMethods.add(method);
+                    }
+                }
+
+                switch (matchingMethods.size()) {
+                    case 0:
+                        currentClass = currentClass.getSuperclass();
+                        break;
+                    case 1:
+                        return matchingMethods.get(0);
```

Comment:
:pick: `case 1:` is not needed (it is covered by the default branch). Therefore an `if then else` may be more appropriate than a `switch case`.